src/inference/rejection.js

```diff
@@ -22,6 +21,17 @@ module.exports = function(env) {
       maxScore: 0,
       incremental: false
     });
+
+    this.throwOnError = options.throwOnError !== undefined ? options.throwOnError : true;
+    // the value of options.probe is the min sample rate
+    this.probe = options.probe;
+    if (this.probe) {
+      // not throw error in probe mode
+      this.throwOnError = false;
+    }
+    this.numSamplesBak = options.samples;
```

Comment:
I was a little confused by the way we track the number of samples that have been taken so far. This is largely down to the existing code in `Rejection`, but I think calling `numSamplesBak` something like `numSamplesTotal` might have helped a little. What do you think?